socs/agents/meinberg_syncbox/agent.py

```diff
@@ -0,0 +1,513 @@
+                          other(144),
+                          internalOscillator(160)
+            mbgSyncboxN2XPtpGrandmasterClockAccuracy::
+                Options:: accurateToWithin25ns(32),
+                          accurateToWithin100ns(33),
+                          accurateToWithin250ns(34),
+                          accurateToWithin1us(35),
+                          accurateToWithin2Point5us(36),
+                          accurateToWithin10us(37),
+                          accurateToWithin25us(38),
+                          accurateToWithin100us(39),
+                          accurateToWithin250us(40),
+                          accurateToWithin1ms(41),
+                          accurateToWithin2Point5ms(42),
+                          accurateToWithin10ms(43),
+                          accurateToWithin25ms(44),
+                          accurateToWithin100ms(45),
+                          accurateToWithin250ms(46),
+                          accurateToWithin1s(47),
+                          accurateToWithin10s(48),
+                          accurateToGreaterThan10s(49)
+        """
+
+        session.set_status('running')
+        self.is_streaming = True
+        while self.is_streaming:
+            yield dsleep(1)
+            if not self.connected:
+                self.log.error('No SNMP response. Check your connection!')
+                self.log.info('Trying to reconnect.')
+
+            read_time = time.time()
+
+            # Check if sample period has passed before getting status
+            if (read_time - self.lastGet) < self.sample_period:
+                continue
+
+            main_get_list = []
+            get_list = []
+
+            # Create the list of OIDs to send get commands
+            oids = ['mbgSyncboxN2XSerialNumber',
+                    'mbgSyncboxN2XFirmwareRevision',
+                    'mbgSyncboxN2XSystemTime',
+                    'mbgSyncboxN2XCurrentRefSource',
+                    'mbgSyncboxN2XPtpProfile',
+                    'mbgSyncboxN2XPtpNwProt',
+                    'mbgSyncboxN2XPtpPortState',
+                    'mbgSyncboxN2XPtpDelayMechanism',
+                    'mbgSyncboxN2XPtpDelayRequestInterval',
+                    'mbgSyncboxN2XPtpTimescale',
+                    'mbgSyncboxN2XPtpUTCOffset',
+                    'mbgSyncboxN2XPtpLeapSecondAnnounced',
+                    'mbgSyncboxN2XPtpGrandmasterClockID',
+                    'mbgSyncboxN2XPtpGrandmasterTimesource',
+                    'mbgSyncboxN2XPtpGrandmasterPriority1',
+                    'mbgSyncboxN2XPtpGrandmasterClockClass',
+                    'mbgSyncboxN2XPtpGrandmasterClockAccuracy',
+                    'mbgSyncboxN2XPtpGrandmasterClockVariance',
+                    'mbgSyncboxN2XPtpOffsetToGrandmaster',
+                    'mbgSyncboxN2XPtpMeanPathDelay']
+
+            for oid in oids:
+                main_get_list.append(('MBG-SYNCBOX-N2X-MIB', oid, 0))
```

Comment:
The string "MBG-SYNCBOX-N2X-MIB" is a constant; declare it at global / class / instance level and reuse.